tools/codegen/api/cpp.py

```diff
@@ -191,9 +195,89 @@ def argument(a: Union[Argument, TensorOptionsArguments, ThisArgument]) -> CppArg
     else:
         assert_never(a)
 
-def group_arguments(
-    func: FunctionSchema, *, method: bool = False
-) -> Sequence[Union[Argument, TensorOptionsArguments, ThisArgument]]:
+@dataclass(frozen=True)
+class CppSignature:
```

Comment:
If any of the other `tools.codegen.api` modules want to make use of these types it will be much easier to put it in `tools.codegen.api.types` as you will be able to avoid import cycles this way.

Reply:
I don't agree. Yes, putting everything into one module avoids import cycles, but it also makes things much less structured. Note also that `CppSignature` here is not actually part of the model. It's something computed from the model that only lives temporarily while the corresponding operator is generated.